apps/web/src/components/invoice-generator/index.tsx

```diff
@@ -0,0 +1,478 @@
+  elementName,
+  formState,
+}: {
+  children: ReactNode;
+  elementName: string;
+  formState: InvoiceGeneratorFormState | null;
+}) {
+  return (
+    <div>
+      {children}
+      {formState?.errors?.[elementName] ? (
+        <span aria-live="polite" className="block text-red-600">
+          {formState.errors[elementName]}
+        </span>
+      ) : null}
+    </div>
+  );
+}
+
+function InvoiceItem({
+  state,
+  index,
+}: {
+  state: InvoiceGeneratorFormState | null;
+  index: number;
+}) {
+  const t = useScopedI18n("invoicegenerator");
+
+  return (
+    <fieldset>
+      <ErrorMessageBlock
+        elementName={`rows[${index.toString()}].product`}
+        formState={state}
+      >
+        <InputRow
+          label={t("Product")}
+          name="rows.product"
+          id={`rows[${index.toString()}].product`}
+          maxLength={128}
+          required
+        />
+      </ErrorMessageBlock>
+      <ErrorMessageBlock
+        elementName={`rows[${index.toString()}].quantity`}
+        formState={state}
+      >
+        <InputRow
+          label={t("Quantity")}
+          name="rows.quantity"
+          id={`rows[${index.toString()}].quantity`}
+          type="number"
+          required
+        />
+      </ErrorMessageBlock>
+      <ErrorMessageBlock
+        elementName={`rows[${index.toString()}].unit`}
+        formState={state}
+      >
+        <InputRow
+          label={t("Unit")}
+          name="rows.unit"
+          id={`rows[${index.toString()}].unit`}
+          defaultValue="kpl"
+          maxLength={128}
+          required
+        />
+      </ErrorMessageBlock>
+      <ErrorMessageBlock
+        elementName={`rows[${index.toString()}].unit_price`}
+        formState={state}
+      >
+        <span>
+          <InputRow
+            label={t("Unit price")}
+            type="number"
+            name="rows.unit_price"
+            id={`rows[${index.toString()}].unit_price`}
+            required
+            step={0.01}
+            min={0}
+            unit="€"
+          />
+        </span>
+      </ErrorMessageBlock>
+    </fieldset>
+  );
+}
+
+function AttachmentRow({
+  state,
+  index,
+}: {
+  state: InvoiceGeneratorFormState | null;
+  index: number;
+}) {
+  const t = useScopedI18n("invoicegenerator");
+
+  return (
+    <fieldset>
+      <ErrorMessageBlock
+        elementName={`attachment_descriptions[${index.toString()}]`}
+        formState={state}
+      >
+        <TextAreaInputRow
+          label={t("Description")}
+          id={`attachment_descriptions[${index.toString()}]`}
+          name="attachment_descriptions"
+          maxLength={512}
+          required
+        />
+      </ErrorMessageBlock>
+      <ErrorMessageBlock
+        elementName={`attachments[${index.toString()}]`}
+        formState={state}
+      >
+        <InputRow
+          label={t("Attachment")}
+          id={`attachments[${index.toString()}]`}
+          name="attachments"
+          type="file"
```

Comment:
Good enough for MVP, but we could enhance the file uploads in the future with file previews, drag n drop etc.